### ISSUE TITLE
ValueError: End location must be greater than or equal to start location when reading SnapGene circular map

### ISSUE THREAD

**Issue by nicolazilio** [biopython/biopython#3181]
### Setup

I am reporting a problem with Biopython version, Python version, and operating system as follows:

```
>>> import sys; print(sys.version)
3.6.8 (default, May 24 2019, 19:22:50) 
[GCC 6.3.0 20170516]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-4.19.0-9-amd64-x86_64-with-debian-10.4
>>> import Bio; print(Bio.__version__)
1.77
```

### Expected behaviour

SnapGene encodes features that span the start/end of a circular map such that their start coordinates are greater than their end coordinates (see attached example).

### Actual behaviour

BioPython does not like features whose start coordinates are greater than their end coordinates, for example

```
File "/home/lib/python3.6/site-packages/Bio/SeqFeature.py", line 801, in __init__
    f"End location ({self.end}) must be greater than "
ValueError: End location (2) must be greater than or equal to start location (4391)
```

### Steps to reproduce

I have attached a relevant SnapGene file to this message, if you want to try to reproduce the issue.
[example.zip](https://github.com/biopython/biopython/files/4999973/example.zip)

**Comment by peterjc** [comment]
OK, the parser will need to split this into a a join with two fragments (like GenBank/EMBL files use).

May we include this example file within Biopython for redistribution as a test case?

**Comment by nicolazilio** [comment]
Sure

**Comment by evkul** [comment]
Actually i just met the same issue with gcc file. Maybe the right way is to swap end and start in case, when end is less than start? That is abosolutely common case - i met the same things in gcc, xdna, gb and seq (Lasergene) files.

**Comment by peterjc** [comment]
We can't do that because the ``Bio.SeqFeature`` objects are very closely aligned to the INSDC feature table used in GenBank and EMBL.

See https://github.com/biopython/biopython/blob/biopython-177/Bio/GenBank/__init__.py#L272 for how this can be done - but note for SnapGene and other files where the format deliberately uses this style of coordinates, you wouldn't want to complain about it with a warning.

**Comment by evkul** [comment]
Don't sure that i understood you correctly. I have files with reverse notation for locations and i have an error from biopython lib. So i have completely no option to solve this issue, except for writing own parser for that

I will attach few examples of files. (Yeah, i know, that BP is not supporting Lasergene Seq files, but still)

[error_files.zip](https://github.com/biopython/biopython/files/5001489/error_files.zip)

**Comment by nicolazilio** [comment]
I don't want to open a can of worms here :), but I was wondering whether the start coordinate must really be less than the end one. If I convert the map that I shared above to GenBank using SnapGene (File -> Export -> DNA -> Save as type: GenBank - Standard), in the resulting file (attached) the start coordinate of the guilty feature "SP6 promoter" is also greater than the end one

```
     promoter        4392..2
                     /label=SP6 promoter
                     /note="promoter for bacteriophage SP6 RNA polymerase"
```
[example.gb.txt](https://github.com/biopython/biopython/files/5005454/example.gb.txt)

Of course, this could be a bug in SnapGene itself, but I checked the INSDC Feature Table Definition, and I don't think it states anywhere that the start coordinate must be less than the end one. But again, I could be wrong.

**Comment by peterjc** [comment]
I think that is a bug in the SnapGene tool's GenBank format output

**Comment by mdehoon** [comment]
Looks OK to me ... same as what you would see in a GFF file.

**Comment by peterjc** [comment]
But GFF and GenBank/EMBL are quite different.

Anyway, this should be fixed on the master branch thanks to #3187 - independent confirmation testing would be most welcome (please comment here and reopen it if needed).
